guides/shipments/overview-of-shipments.md

```diff
@@ -0,0 +1,463 @@
+  "USPS Global Express Guaranteed Non-Document Non-Rectangular"=>4295,
+  "USPS Global Express Guaranteed Non-Document Rectangular"=>4295,
+  "USPS Global Express Guaranteed (GXG)"=>4295,
+  "USPS Express Mail International"=>2895,
+  "USPS Priority Mail International Small Flat Rate Box"=>1345
+}
+```
+
+From all of the viable shipping services in this hash, the `compute` method selects the one that matches the description of the calculator. At this point, an optional flat handling fee (set via preferences) can be added:
+
+```ruby
+rate = rates[self.description].to_f + (Spree::ActiveShipping::Config[:handling_fee].to_f || 0.0)
+```
+
+Finally, don't forget to register the calculator you added. In extensions, this is accomplished with the `activate` method:
+
+```ruby
+def activate
+  Calculator::Usps::FirstClassMailInternationalParcels.register
+end
+```
+
+## Filtering Shipping Methods On Criteria Other Than the Zone
+
+Ordinarily, it is the zone of the shipping address that determines which shipping methods are displayed to a customer at checkout. Here is how the availability of a shipping method is determined:
+
+```ruby
+class Spree::Stock::Estimator
+  def shipping_methods(package)
+    shipping_methods = package.shipping_methods
+    shipping_methods.delete_if { |ship_method| !ship_method.calculator.available?(package.contents) }
+    shipping_methods.delete_if { |ship_method| !ship_method.include?(order.ship_address) }
+    shipping_methods.delete_if { |ship_method| !(ship_method.calculator.preferences[:currency].nil? || ship_method.calculator.preferences[:currency] == currency) }
+    shipping_methods
+  end
+end
+```
+
+Unless overridden, the calculator's `available?` method returns `true` by default. It is, therefore, the zone of the destination address that filters out the shipping methods in most cases. However, in some circumstances it may be necessary to filter out additional shipping methods.
+
+Consider the case of the USPS First Class domestic shipping service, which is not offered if the weight of the package is greater than 13oz. Even though the USPS API does not return the option for First Class in this instance, First Class will appear as an option in the checkout view with an unfortunate value of 0, since it has been set as a Shipping Method.
+
+To ensure that First Class shipping is not available for orders that weigh more than 13oz, the calculator's `available?` method must be overridden as follows:
+
+```ruby
+class Calculator::Usps::FirstClassMailParcels < Calculator::Usps::Base
+  def self.description
+    "USPS First-Class Mail Parcel"
+  end
+
+  def available?(order)
+    multiplier = 1.3
+    weight = order.line_items.inject(0) do |weight, line_item|
+      weight + (line_item.variant.weight ? (line_item.quantity - line_item.variant.weight - multiplier) : 0)
+    end
+    #if weight in ounces > 13, then First Class Mail is not available for the order
+      weight > 13 ? false : true
+  end
+end
```

Comment:
I'd use a code example with something that is not included into an extension (Usps stuff), so that it needs less overhead and we can keep it simpler.